torch/nn/functional.py

```diff
@@ -3043,10 +3047,13 @@ def cross_entropy(
             reduce=reduce,
             reduction=reduction,
             label_smoothing=label_smoothing,
+            dim=dim,
         )
     if size_average is not None or reduce is not None:
         reduction = _Reduction.legacy_get_string(size_average, reduce)
-    return torch._C._nn.cross_entropy_loss(input, target, weight, _Reduction.get_enum(reduction), ignore_index, label_smoothing)
+        if dim is not None:
+            input = input.swapaxes(dim, 1)
+    return torch._C._nn.cross_entropy_loss(input, target, weight, _Reduction.get_enum(reduction), ignore_index, label_smoothing, dim)
```

Comment:
Passing `dim` here isn't going to work. This PR doesn't change the interface of the underlying ATen op exposed via `torch._C._nn.cross_entropy_loss` (and I don't think it should).

Rather, we'll need to swap axes back for the output when we're not doing a mean or sum reduction.

Reply:
@jbschlosser, I've removed `dim` from here, but I couldn't understand how to swap the axes back when we're not doing a mean or sum reduction.

Reply:
something like this should work:
```python
output = torch._C._nn.cross_entropy_loss(input, target, weight, _Reduction.get_enum(reduction), ignore_index, label_smoothing, dim)
if reduction == 'none':
    output = output.swapaxes(dim, 1)
return output
```

this is untested but the tests you wrote should verify it :)

Reply:
Thanks, I've added this